lib/land/trackers/api_tracker.rb

```diff
@@ -206,24 +227,69 @@ def user_agent
                      Land.config.blank_user_agent_string
 
         @user_agent = UserAgent[user_agent]
+        @user_agent.user_agent_type = UserAgentType['user']
 
         if Land.config.identify_crawlers && defined?(CrawlerDetect)
           crawler_detect = CrawlerDetect.new(user_agent)
-          user_agent_type = crawler_detect.is_crawler? ? 'crawl' : 'api'
-          @user_agent.update(user_agent_type: UserAgentType[user_agent_type])
+          @user_agent.user_agent_type = UserAgentType['crawl'] if crawler_detect.is_crawler?
         end
 
         browser = ::Browser.new(user_agent)
 
-        @user_agent.update(
-          browser: Browser[browser.name],
-          device: Device[browser.device.name],
-          platform: Platform[browser.platform.name],
-          browser_version: browser.version
-        )
+        update_browser_color_preferences(Browser[browser.name])
+
+        @user_agent.browser = Browser[browser.name]
+        @user_agent.device = Device[browser.device.name]
+        @user_agent.platform = Platform[browser.platform.name]
+        @user_agent.browser_version = browser.version
+        @user_agent.device_resolution = device_resolution
```

Comment:
This should make this more clear

```suggestion
        land_browser = Land::Browser[browser.name]
        update_browser_color_preferences(land_browser)

        @user_agent.browser = land_browser
        @user_agent.device = Device[browser.device.name]
        @user_agent.platform = Platform[browser.platform.name]
        @user_agent.browser_version = browser.version
        @user_agent.device_resolution = device_resolution
```